packages/toolkit/src/query/fetchBaseQuery.ts

```diff
@@ -54,7 +54,7 @@ const handleResponse = async (
 
   if (responseHandler === 'json') {
     const text = await response.text()
-    return text.length ? JSON.parse(text) : undefined
+    return text.length ? JSON.parse(text) : null
```

Comment:
Note that this is a almost-breaking change, but a very minor one.

It is necessary as we cannot serialize `undefined` when transferring data from server to client in SSR scenarios.